src/MVVM/Factories/ViewModelFactory.cs

```diff
@@ -121,19 +204,62 @@ internal static INavigationService TryResolveNavigationServiceInstance()
         }
 
 
+
+        /// <summary>
+        /// Creates a new ViewModel instance and invokes the <see cref="IViewModel{TModel}.InitializeAsync(TModel)"/> method.
+        /// </summary>
+        /// <typeparam name="TViewModel">Type of the ViewModel to create</typeparam>
+        /// <typeparam name="TModel">Type of the parameter that will be passed in the <see cref="IViewModel{TModel}.InitializeAsync(TModel)"/> method</typeparam>
+        /// <returns>A ViewModel instance of the given type</returns>
+        public static async Task<TViewModel> ResolveAsync<TViewModel, TModel>(
+            TModel model)
+            where TViewModel : class, IViewModel<TModel>
+        {
+            try
+            {
+                var instance = iocContainer.Resolve<TViewModel>();
+
+                await instance.InitializeAsync(
+                    model);
+
+                return instance;
+            }
+            catch (Exception innerException)
+            {
+                string errorMessage = $"ViewModel of type {typeof(TViewModel).Name} cannot be resolved --- is it registered?";
+
+                log?.Critical(
+                    errorMessage,
+                    innerException);
+
+                throw new TypeLoadException(
+                    errorMessage,
+                    innerException);
+            }
+        }
+
         /// <summary>
-        /// Creates a new ViewModel instance, invokes the InitializeAsync method using the parameter and returns the initialized instance
+        /// Creates a new ViewModel instance and invokes the <see cref="IViewModel{TModel}.InitializeAsync(TModel)"/> method.
         /// </summary>
-        /// <typeparam name="TViewModelInterface">Type of the ViewModel to create</typeparam>
-        /// <typeparam name="TModel">Type of the parameter that will be used for initialization</typeparam>
-        /// <returns>ViewModel instance of the given type</returns>
-        public static async Task<TViewModelInterface> ResolveAsync<TViewModelInterface, TModel>(
+        /// <typeparam name="TModel">Type of the parameter that will be passed in the <see cref="IViewModel{TModel}.InitializeAsync(TModel)"/> method</typeparam>
+        /// <param name="viewModelType"></param>
+        /// <param name="model">Parameter that will be passed in the <see cref="IViewModel{TModel}.InitializeAsync(TModel)"/> method</param>
```

Comment:
Codacy found an issue: [Split this 140 characters long line (which is greater than 130 authorized).](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5767060)